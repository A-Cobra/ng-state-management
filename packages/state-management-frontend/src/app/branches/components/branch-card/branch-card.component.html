<clapp-card>
  <clapp-card-header>
    <h5>{{ branch.name }}</h5>
  </clapp-card-header>
  <clapp-card-content>
    <img
      width="200"
      [src]="branch.image ?? 'assets/img/placeholder-image.png'"
      [alt]="branch.name"
    />
  </clapp-card-content>
  <clapp-card-footer>
    <p>{{ branch.schedule }}</p>
  </clapp-card-footer>
</clapp-card>
