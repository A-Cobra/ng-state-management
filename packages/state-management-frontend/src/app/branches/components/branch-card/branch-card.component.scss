clapp-card {
  &:hover {
    cursor: pointer;
  }

  img {
    width: 100%;
  }
}
