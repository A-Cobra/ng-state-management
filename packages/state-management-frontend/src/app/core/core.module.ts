import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { PhoneNumberDirective } from './directives/phone-number.directive';
import { FloatNumberOrNumberRangeDirective } from './directives/float-number-or-number-range.directive';

import {
  ClappButtonModule,
  ClappTextInputModule,
  ClappInputHelpersModule,
} from '@clapp1/clapp-angular';

import { ModalGoBackComponent } from './components/modal-go-back/modal-go-back.component';
import { ReactiveFormControlTextInputComponent } from './components/reactive-form-control-text-input/reactive-form-control-text-input.component';
import { FullErrorNamePipe } from './pipes/full-error-name.pipe';

@NgModule({
  imports: [
    CommonModule,
    ReactiveFormsModule,
    ClappButtonModule,
    ClappTextInputModule,
    ClappInputHelpersModule,
  ],
  declarations: [
    PhoneNumberDirective,
    FloatNumberOrNumberRangeDirective,
    ReactiveFormControlTextInputComponent,
    FullErrorNamePipe,
    ModalGoBackComponent,
  ],
  exports: [
    PhoneNumberDirective,
    FloatNumberOrNumberRangeDirective,
    ReactiveFormControlTextInputComponent,
    FullErrorNamePipe,
  ],
  schemas: [CUSTOM_ELEMENTS_SCHEMA],
})
export class CoreModule {}
