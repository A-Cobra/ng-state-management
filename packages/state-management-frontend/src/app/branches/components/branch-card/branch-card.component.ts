import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { Branch } from '../../models/branch.model';

@Component({
  selector: 'app-branch-card',
  templateUrl: './branch-card.component.html',
  styleUrls: ['./branch-card.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class BranchCardComponent {
  @Input() branch: Branch;
}
