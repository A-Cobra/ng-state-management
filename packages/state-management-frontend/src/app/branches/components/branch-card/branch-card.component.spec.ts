import { ComponentFixture, TestBed } from '@angular/core/testing';

import { BranchCardComponent } from './branch-card.component';
import { MOCK_BRANCH } from '../../test/mocks';

import { ClappCardModule } from '@clapp1/clapp-angular';

describe('BranchCardComponent', () => {
  let component: BranchCardComponent;
  let fixture: ComponentFixture<BranchCardComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [BranchCardComponent],
      imports: [ClappCardModule],
    }).compileComponents();

    fixture = TestBed.createComponent(BranchCardComponent);
    component = fixture.componentInstance;
    component.branch = MOCK_BRANCH;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render branch information', () => {
    expect(fixture).toMatchSnapshot();
  });
});
