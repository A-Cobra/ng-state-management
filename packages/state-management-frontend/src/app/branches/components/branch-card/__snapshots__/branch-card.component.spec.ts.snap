// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BranchCardComponent should render branch information 1`] = `
<state-management-app-branch-card
  branch={[Function Object]}
>
  <clapp-card>
    <div
      class="clapp-card-wrapper"
    >
      <clapp-card-content>
        <div
          class="clapp-card-content-wrapper"
        >
          <img
            alt="Mock branch"
            height="228"
            src="https://dummyimage.com/400x400/000000/fff"
          />
        </div>
      </clapp-card-content>
      <clapp-card-header>
        <div
          class="clapp-card-header-wrapper"
        >
          <h3>
            <h5>
              Mock branch
            </h5>
          </h3>
        </div>
      </clapp-card-header>
      <clapp-card-footer>
        <div
          class="clapp-card-footer-wrapper"
        >
          <p>
            10:00 - 19:00
          </p>
        </div>
      </clapp-card-footer>
    </div>
  </clapp-card>
</state-management-app-branch-card>
`;
